### PR TITLE
Cmfd output

### PR DESCRIPTION
**Pull request by shikhar413**
This PR addresses issue #961 to output CMFD parameters correctly. Initially what was happening was that all output was being displayed to output buffer in method finalize_generation in output.F90. However, CMFD was not called by then and so CMFD parameters would not be set and display as 0.

To fix this, a separate "print_cmfd" routine was created and called during finalize_batch, after CMFD has actually finished running. print_cmfd outputs necessary CMFD output and prints a new line to ensure consistent formatting. The new line is printed regardless of whether CMFD is on or not.

PR can be tested on CMFD test functions, to see that CMFD output is correct and not printing just 0's anymore.

Some things that might require some further discussion/modification:
  1) Naming of print_cmfd routine - Should it be something more general to signify output of any batch-level parameters?
  2) Placement of print_cmfd routine - Currently called right after execute_cmfd, should routine be placed later?
  3) Code conventions - Do let me know if there are any code conventions I might have missed

### CODE REVIEW

File: src/output.F90 [hunk 1]
```diff
@@ -376,6 +376,16 @@ subroutine print_batch_keff()
     else
       write(UNIT=OUTPUT_UNIT, FMT='(23X)', ADVANCE='NO')
     end if
+
```
Comment (paulromano) [inline src/output.F90]: Remove trailing whitespace here

File: src/output.F90 [hunk 2]
```diff
@@ -376,6 +376,16 @@ subroutine print_batch_keff()
     else
       write(UNIT=OUTPUT_UNIT, FMT='(23X)', ADVANCE='NO')
     end if
+      
+  end subroutine print_batch_keff
+
+!===============================================================================
+! PRINT_CMFD displays the CMFD related information to output after CMFD is
```
Comment (paulromano) [inline src/output.F90]: Remove trailing whitespace

File: src/output.F90 (hunk 1 above)
Comment (paulromano) [inline src/output.F90]: Remove trailing whitespace

File: src/output.F90 [hunk 3]
```diff
@@ -376,6 +376,16 @@ subroutine print_batch_keff()
     else
       write(UNIT=OUTPUT_UNIT, FMT='(23X)', ADVANCE='NO')
     end if
+      
+  end subroutine print_batch_keff
+
+!===============================================================================
+! PRINT_CMFD displays the CMFD related information to output after CMFD is 
+! executed. Will print blank line if CMFD is not on, to ensure consistent
```
Comment (paulromano) [inline src/output.F90]: Remove trailing whitespace

File: src/output.F90 [hunk 4]
```diff
@@ -376,6 +376,16 @@ subroutine print_batch_keff()
...
+      
+  end subroutine print_batch_keff
+
+!===============================================================================
+! PRINT_CMFD displays the CMFD related information to output after CMFD is 
+! executed. Will print blank line if CMFD is not on, to ensure consistent 
+! formatting of output columns
+!===============================================================================
+
+  subroutine print_cmfd()
```
Comment (paulromano) [inline src/output.F90]: There's really no need to split up the `print_batch_keff` subroutine. Instead, I would just have it be called from `finalize_batch` (where you're currently calling `print_cmfd`).

File: src/simulation.F90 [hunk 5]
```diff
@@ -338,6 +339,7 @@ subroutine finalize_batch()
     if (run_mode == MODE_EIGENVALUE) then
       ! Perform CMFD calculation if on
       if (cmfd_on) call execute_cmfd()
+      call print_cmfd()
```
Comment (paulromano) [inline src/simulation.F90]: So here, you could have:
```Fortran
if (master .and. verbosity > 7) call print_batch_keff()
```

File: src/output.F90 [hunk 6]
```diff
@@ -376,7 +376,7 @@ subroutine print_batch_keff()
     else
       write(UNIT=OUTPUT_UNIT, FMT='(23X)', ADVANCE='NO')
     end if
-
+
```
Comment (paulromano) [inline src/output.F90]: Extraneous whitespace here again

File: src/output.F90 [hunk 7]
```diff
@@ -356,7 +356,7 @@ subroutine print_batch_keff()
     integer :: n  ! number of active generations
 
     ! Determine overall generation and number of active generations
-    i = overall_generation()
+    i = overall_generation() - 1
```
Comment (paulromano) [inline src/output.F90]: Rather than relying on overall_generation here, we should probably just make this:
```Fortran
i = current_batch*gen_per_batch
```

Comment (shikhar413) [reply src/output.F90]: Thanks @paulromano, pushed some changes without realizing they would update the PR as well. I'm still in the middle of figuring out some issues with travis but will push some more commits once that is done.

Comment (paulromano) [reply src/output.F90]: Yeah, I noticed a few tests were failing. The change I'm suggesting here should hopefully fix those tests.

**Comment by paulromano** [comment]
Thanks again for the fix, @shikhar413!

**Comment by shikhar413** [comment]
No worries, thanks for the help!

**Pull request merged**